src/Library/demos/Drag and Drop/main.js

```diff
@@ -0,0 +1,60 @@
+import Gtk from "gi://Gtk";
+import Adw from "gi://Adw";
+import Gdk from "gi://Gdk";
+import GObject from "gi://GObject";
+
+const list = workbench.builder.get_object("list");
+
+// Calculate the length of our ListBox
+let last_row = list.get_last_child();
+let list_length = last_row.get_index() + 1;
+
+// Create a Dragsource and DropTarget for each row
+for (let i = 0; i < list_length; i++) {
+  let row = list.get_row_at_index(i);
+  row.selectable = false;
+  let _drag_x;
+  let _drag_y;
+
+  let allocation;
+  const drag = new Gtk.DragSource();
```

Comment:
```suggestion
  const drag = new Gtk.DragSource({
    actions: Gdk.DragAction.MOVE,
  });
```

Reply:
We use snake case for variables. I'd suggest  `drag_source`
the main reasoning is that it makes it easy to differentiate quickly from a GI class